### PR TITLE
Implement base for battle system

### PR DESCRIPTION
**Pull request by guimcaballero**
This PR adds a basic interactable component, which just changes the state into Battle when the player walks into it. This is done by checking the distance between all players and all interactibles, so it's technically `O(n^2)` and we should come up with a better way to deal with it.

We probably want to add something similar to what we have on the buttons to decide what each interactable does.

The Battle system spawns a square (no hexagons for now, it's not built in) board. Pressing `ESC` during Battle changes state to Overworld. 

The Overworld camera is being used for Battle, as it wasn't being despawned, and we pretty much want the same behavior for now. When the issue with despawning the camera is fixed, we should probably have a separate camera be spawned in Battle.

We also probably need a way to keep track of where the player was, so it can appear in the same place when going back to the Overworld.

This code is just laying the basics, I expect everything to be changed.

Video:

https://user-images.githubusercontent.com/29580620/107996825-9438b080-6fe1-11eb-9be7-f0daa14ee65b.mov

### CODE REVIEW

**Comment by guimcaballero** [comment]
Hexagons are a go!

Also, AmbientLights have been changed for LightBundles, as they provide better looking lighting.

https://user-images.githubusercontent.com/29580620/107999280-5b9bd580-6fe7-11eb-8a6e-7e97266339a6.mov

File: src/menu/mod.rs [hunk 1]
```diff
@@ -64,6 +64,7 @@ impl Plugin for MenuPlugin {
     }
 }
 
+#[allow(dead_code)]
```
Comment (Byteron) [inline src/menu/mod.rs]: It's not necessary, CI will pass anyways.
I'd remove it for now.

**Review by Byteron — CHANGES_REQUESTED** [review]
one minor request, otherwise it looks good to me.

File: src/menu/mod.rs (hunk 1 above)
Comment (Ratysz) [reply src/menu/mod.rs]: Confusion like this is seriously tempting me to remove that allowance again 😄

File: src/overworld/interactable.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+use crate::AppState;
+
+pub struct Interactable;
+
+pub fn interact_with_interactables(
+    interactables: Query<&Transform, With<Interactable>>,
+    player: Query<&Transform, With<Player>>,
+    mut state: ResMut<State<AppState>>,
+) {
+    // NOTE: O(n^2), should be changed for a smarter implementation
```
Comment (Ratysz) [inline src/overworld/interactable.rs]:
```suggestion
    // TODO: O(n^2), should be changed for a smarter implementation
```

File: src/battle/mod.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+use crate::APPSTATES;
+
+/// Marker for despawning when exiting `AppState::Battle`
+pub struct StateCleanup;
+
+pub struct BattlePlugin;
+impl Plugin for BattlePlugin {
+    fn build(&self, app: &mut AppBuilder) {
+        app.on_state_enter(APPSTATES, AppState::Battle, setup_battle.system())
+            // NOTE: Uses same camera as overworld
```
Comment (Ratysz) [inline src/battle/mod.rs]:
```suggestion
            // TODO: Uses same camera as overworld
```

`TODO` gets picked by most IDEs worth their salt. `NOTE` is rarer.

**Review by Ratysz — APPROVED** [review]

**Review by Ratysz — CHANGES_REQUESTED** [review]
(Don't mind me, just stumbling on a github UX pothole.)

**Comment by TheRawMeatball** [comment]
> This is done by checking the distance between all players and all interactibles, so it's technically `O(n^2)`

I feel like this is exaggerated: this is right now `O(p*i)` for # players and # interactibles, but since there's only ever one player this is just linear time - it's not the best either, but we can fix it when we integrate a collision solution

File: src/menu/mod.rs (hunk 1 above)
Comment (Byteron) [inline src/menu/mod.rs]:
```suggestion
```

**Review by Ratysz — APPROVED** [review]

**Pull request merged**